Assets/Scripts/Minigames/CowardMiniGame/RipcordBehavior.cs

```diff
@@ -12,216 +12,243 @@
-        {
-            _currentLight = 0;
-            StartCoroutine(blinkingLight());
-            _lightDoOnce = false;
-        }
-        if (_numReleased == 1 && _gameStarted == true && _lightDoOnce == true)
+
+        // Make sure the score detection range is destroyed once the ripcord returns to the start position
+        if(_scoreDetectionRange != null)
         {
-            _listLights[0].SetActive(true);
-            _currentLight = 1;
-            StartCoroutine(blinkingLight());
-            _lightDoOnce = false;
+            Destroy(_scoreDetectionRange);
         }
-        if(_numReleased == 2 && _gameStarted == true && _lightDoOnce == true)
+    }
+
+    /// <summary>
+    /// When the player successfully scores
+    /// </summary>
+    private void OnScore()
+    {
+        if (_canScore)
         {
-            _listLights[1].SetActive(true);
-            _currentLight = 2;
-            StartCoroutine(blinkingLight());
-            _lightDoOnce = false;
+            _score++;
+
+            OnRipcordScore?.Invoke();
+            _successfulPulls.text = (_score).ToString();
         }
-        if (_numReleased == 3)
+
+        // Winning score
+        if (_score >= 3)
         {
-            _listLights[2].SetActive(true);
-            StopCoroutine(blinkingLight());
             _gears.SetActive(true);
             _successfulPulls.color = Color.green;
-            StopRipcordSteam();
-            //Destroy(this);
-        }
-        //Debug.Log(_numReleased);
+            StopGeneratorSteam();
 
+            DeactivateMinigame();
+        }
     }
 
     /// <summary>
     /// Stops the steam coming from the ripcord.
     /// </summary>
-    public void StopRipcordSteam()
+    public void StopGeneratorSteam()
     {
         _steam.Stop();
     }
 
-    IEnumerator PauseEPress()
-    {
-        yield return new WaitForSeconds(0.4f);
-        _switchDoOnce = true;
-    }
-    IEnumerator blinkingLight()
-    {
-        _listLights[_currentLight].SetActive(true);
-        yield return new WaitForSeconds(_lightSpeed);
-        _listLights[_currentLight].SetActive(false);
-        _lightDoOnce = true;
-    }
-    IEnumerator ReleaseWindow()
-    {
-        _hasReleased = true;
-        yield return new WaitForSeconds(0.2f);
-        _hasReleased = false;
-    }
-    IEnumerator FindPreview()
-    {
-        _preview = GameObject.FindWithTag("Release");
-        yield return _preview;
-    }
-    void OnTriggerEnter(Collider col)
+    void OnTriggerEnter(Collider collider)
     {
-        if (col.gameObject.tag == "Release" && _hasReleased == true)
+        if (_isMinigameActive)
         {
-            _numReleased++;
-            GameObject.Destroy(col.gameObject);
-        }
+            // Enable player interaction
+            if (collider.gameObject.tag == "Player")
+            {
+                _playerController.Interact.started += OnInteract;
+                TabbedMenu.Instance.ToggleInteractPrompt(true, "GRAB");
```

Comment:
Ideally the interaction prompt will display only when the player is looking at the ripcord, not just when they are in the trigger box. To do this you would need to add a collider to the ripcord (not a trigger) and implement the IInteractable interface.
This may be more difficult than it is worth, as in my experience game objects dont like having multiple colliders, and this would probably involve adding a child game object on the ripcord with the collider and interaction prompts listed there. As well as a bool on that child object to show the ripcord script when the player is looking at it and it is interactable, and when it is not.

Reply:
https://discord.com/channels/1237040972920262756/1264778363177009354/1265484511463080067